scripts/deployment/build-and-deploy.sh

```
@@ -0,0 +1,700 @@
+    printf '%s\n' "$current_fingerprint" > "$INSTALL_FINGERPRINT_FILE"
+}
+
+# =============================================================================
+# Lock Verification
+# =============================================================================
+lock_require_or_exit() {
+    if [ ! -f "$LOCK_FILE" ]; then
+        log_error "No deployment lock. Run via: make ship, make staging, or make production"
+        exit 1
+    fi
+    local lock_pid=$(cut -d'|' -f1 "$LOCK_FILE" 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)